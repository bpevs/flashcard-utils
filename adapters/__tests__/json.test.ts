import { assertSnapshot } from 'jsr:@std/testing/snapshot'
import fromJSON from '../from_json.ts'
import toJSON from '../to_json.ts'
import deckData from './__data__/zh_CN.ts'

Deno.test('init Deck to/from JSON', async (t) => {
  const deck = fromJSON(JSON.stringify(deckData))
  await assertSnapshot(t, toJSON(deck))
})
