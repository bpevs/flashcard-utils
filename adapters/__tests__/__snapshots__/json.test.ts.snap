export const snapshot = {};

snapshot[`init Deck to/from JSON 1`] = `
[32m'{
  "id": "zh-CN",
  "idNum": 1221044567,
  "name": "中文（普通话）",
  "desc": "Flashcards for Chinese",
  "meta": {
    "name_en": "Chinese (Mandarin)",
    "name_short": "中文",
    "language_code": "zh",
    "locale_code": "zh-CN",
    "locale_code_azure": "zh-CN",
    "locale_code_deepl": "zh",
    "locale_flag": "🇨🇳",
    "voice_id_azure": "zh-CN-XiaoxiaoNeural"
  },
  "content": {
    "fields": ["category", "emoji", "text", "pinyin", "sound"],
    "watch": ["emoji", "text", "pinyin"]
  },
  "notes": [
    ["animal", "🐈", "猫", "māo", "[sound:zh-cn_猫.mp3]"],
    ["animal", "🐶", "狗", "gǒu", "[sound:zh-cn_狗.mp3]"],
    ["body", "🦶", "脚", "jiǎo", "[sound:zh-cn_脚.mp3]"],
    ["color", "🟢", "绿", "lǜ", "[sound:zh-cn_.mp3]"],
    ["directions", "⬇️🍂", "下", "xià", "[sound:zh-cn_下.mp3]"],
    ["food", "🧊", "冰", "bīng", "[sound:zh-cn_冰.mp3]"],
    ["number", "7️⃣", "七", "qī", "[sound:zh-cn_七.mp3]"],
    ["number", "8️⃣", "八", "bā", "[sound:zh-cn_八.mp3]"],
    ["number", "9️⃣", "九", "jiǔ", "[sound:zh-cn_九.mp3]"],
    ["transportation", "🚗", "汽车", "qì chē", "[sound:zh-cn_汽车.mp3]"],
    ["verbs", "❤️😍", "爱", "ài", "[sound:zh-cn_爱.mp3]"]
  ]
}'[39m
`;
